### ISSUE TITLE
Allow implicit 'this' with destructured param

### ISSUE THREAD

**Issue by scronay** [gajus/eslint-plugin-jsdoc#1190]
## Expected behavior

No warning should be triggered when `this` param is accompanied by a destructured parameter.

## Actual behavior

A warning is triggered, implying that the `@param` for the destructured root is being referenced as a match for `this`

![image](https://github.com/gajus/eslint-plugin-jsdoc/assets/10538620/de077619-a5fa-4b2d-b623-d9717c6d912f)

**Note**: this warning does not trigger when the parameter following `this` is not destructured, or when `@param` is explicitly provided for `this`, as in the following cases:

```js
/**
 * Returns the sum of two numbers
 * @param {number} a First value
 * @param {number} b Second value
 * @returns {number} Sum of a and b
 */
function sum(this: unknown, a: number, b: number) {
	return a + b;
}
```

```js
/**
 * Returns the sum of two numbers
 * @param this 'this'
 * @param options Object to destructure
 * @param options.a First value
 * @param options.b Second value
 * @returns Sum of a and b
 */
function sumDestructure(this: T, { a, b }: { a: number, b: number }) {
	return a + b;
}
```

## ESLint Config

```js
module.exports = {
	env: {
		node: true
	},
	extends: [
		'eslint:recommended',
		'plugin:@typescript-eslint/recommended',
		'plugin:jsdoc/recommended'
	],
	parser: '@typescript-eslint/parser',
	plugins: [
		'@typescript-eslint',
		'jsdoc'
	],
	rules: {
		'jsdoc/require-param-type': 0,
		'jsdoc/require-returns-type': 0
	}
}
```

## ESLint sample

![image](https://github.com/gajus/eslint-plugin-jsdoc/assets/10538620/b40bb1a8-ad6e-4669-9977-5d81cfde38cc)

```js
/**
 * Returns the sum of two numbers
 * @param options Object to destructure
 * @param options.a First value
 * @param options.b Second value
 * @returns Sum of a and b
 */
function sumDestructure(this: unknown, { a, b }: { a: number, b: number }) {
	return a + b;
}
```

## Environment

- Node version: v18.16.0 
- ESLint version 8.46.0 
- `eslint-plugin-jsdoc` version: 48.0.2
